shared/folders/row.desktop.js

```diff
@@ -0,0 +1,183 @@
+// @flow
+import React from 'react'
+import type {Folder} from './list'
+import {Box, Text, Icon, Avatar, Meta} from '../common-adapters'
+import type {Props as IconProps} from '../common-adapters/icon'
+import {globalStyles, globalColors} from '../styles/style-guide'
+import {resolveImageAsURL} from '../../desktop/resolve-root'
+
+const Avatars = ({styles, users}) => (
+  <Box style={styles.avatarContainer}>
+    {users.length === 1 || users.length === 2
+      ? <Avatar size={32} username={users[users.length - 1].username} />
+      : <Icon type={styles.groupIcon} />}
+  </Box>
+)
+
+const Names = ({styles, users}) => {
+  return (
+    <Box style={stylesBodyNameContainer}>
+      {users.map((u, i) => (
+        <Text
+          key={u.username}
+          type={u.you ? 'BodySemiboldItalic' : 'BodySemibold'}
+          style={{color: u.broken ? globalColors.red : styles.nameColor}}>{u.username}
+          {
+            (i !== users.length - 1) && // Injecting the commas here so we never wrap and have newlines starting with a ,
+              <Text type='BodySemibold' style={{color: styles.nameColor, marginRight: 2}}>,</Text>}
+        </Text>
+      ))}
+    </Box>
+  )
+}
+
+const Modified = ({styles, modified}) => (
+  <Box style={stylesModified}>
+    <Icon type='thunderbolt' style={{marginRight: 5}} />
+    <Text type='BodySmall' backgroundMode={styles.modifiedMode}>Modified {modified.when} by&nbsp;</Text>
+    <Text type='BodySmallLink' backgroundMode={styles.modifiedMode}>{modified.username}</Text>
+  </Box>
+)
+
+const RowMeta = ({ignored, meta, styles}) => {
+  const metaColors = {
+    'new': globalColors.white,
+    'rekey': globalColors.white
+  }
+
+  const metaBGColors = {
+    'new': globalColors.blue2,
+    'rekey': globalColors.red
+  }
+
+  const metaProps = ignored
+    ? {title: 'ignored', style: styles.ignored}
+    : {title: meta || '', style: {color: metaColors[meta], backgroundColor: metaBGColors[meta]}}
+
+  return <Meta {...metaProps} />
+}
+
+const Row = ({users, isPublic, ignored, isFirst, meta, modified, hasData}: Folder) => {
+  const styles = isPublic ? stylesPublic : stylesPrivate
+
+  const containerStyle = {
+    ...styles.rowContainer,
+    ...(isFirst && {borderBottom: undefined})}
```

Comment:
Side comment: this is a really interesting pattern with spread operators that I haven't seen before.